test/framework/integration_framework/fake_peer/types.hpp

```diff
@@ -91,7 +91,7 @@ namespace integration_framework {
         std::vector<std::reference_wrapper<const shared_model::proto::Block>>;
     using OrderingProposalRequest = iroha::consensus::Round;
     using OrderingProposalRequestResult =
-        boost::optional<shared_model::proto::Proposal &>;
+        boost::optional<std::shared_ptr<const shared_model::proto::Proposal>>;
```

Comment:
Are there any lifetime issues with using reference? Maybe make a comment in `proposal_storage.hpp` or some better place.